assets/app/views/_volumes.html

```diff
@@ -40,11 +52,38 @@
     </div>
     <div ng-if="volume.gitRepo">
       <dt>Type:</dt>
-      <dd>git repo</dd>
+      <dd>
+        git repo
+       <span class="small text-muted">(a volume that is pulled from git when the pod is created)</span>
+      </dd>
       <dt>Repository:</dt>
       <dd>{{volume.gitRepo.repository}}</dd>
       <dt ng-if-start="volume.gitRepo.revision">Revision:</dt>
       <dd ng-if-end>{{volume.gitRepo.revision}}</dd>
     </div>
+    <div ng-if="volume.downwardAPI">
+      <dt>Type:</dt>
+      <dd>
+        downward API
+        <span class="small text-muted">(a volume populated by information about the pod)</span>
+       </dd>
+      <div ng-repeat="item in volume.downwardAPI.items">
+        <dt>Volume file:</dt>
+        <dd>{{item.fieldRef.fieldPath}}&#8201;&#8594;&#8201;{{item.path}}</dd>
+      </div>
+    </div>
+    <div ng-if="volume.configMap">
+      <dt>Type:</dt>
+      <dd>
+        config map
+        <span class="small text-muted">(a volume populated by a config map)</span>
+       </dd>
+      <dt>Name:</dt>
+      <dd>{{volume.configMap.name}}</dd>
+      <div ng-repeat="item in volume.configMap.items">
+        <dt>Volume file:</dt>
+        <dd>{{item.path}}&#8201;&#8594;&#8201;{{item.key}}</dd>
```

Comment:
heh, yeah... telling you your volume is a volume is a bit redundant, I
assume that describer is coming from k8s?  cc @smarterclayton

so ConfigMap and Secret in the CLI don't even list out the keys to paths?

On Thu, Mar 17, 2016 at 2:39 PM, Fabiano Franz <notifications@github.com>
wrote:

> In assets/app/views/_volumes.html
> <https://github.com/openshift/origin/pull/8073#discussion_r56556723>:
>
> > +      <div ng-repeat="item in volume.downwardAPI.items">
> > +        <dt>Volume file:</dt>
> > +        <dd>{{item.fieldRef.fieldPath}}&#8201;&#8594;&#8201;{{item.path}}</dd>
> > +      </div>
> > +    </div>
> > +    <div ng-if="volume.configMap">
> > +      <dt>Type:</dt>
> > +      <dd>
> > +        config map
> > +        <span class="small text-muted">(a volume populated by a config map)</span>
> > +       </dd>
> > +      <dt>Name:</dt>
> > +      <dd>{{volume.configMap.name}}</dd>
> > +      <div ng-repeat="item in volume.configMap.items">
> > +        <dt>Volume file:</dt>
> > +        <dd>{{item.path}}&#8201;&#8594;&#8201;{{item.key}}</dd>
>
> Here's how they look like in the CLI, for comparison:
>
> Volumes:
>   default-token-x0svm:
>     Type:   Secret (a volume populated by a Secret)
>     SecretName: default-token-x0svm
>   localvol1:
>     Type:   ConfigMap (a volume populated by a ConfigMap)
>     Name:   pod/pod-volume-downward-api
>   localvol2:
>     Type:   DownwardAPI (a volume populated by information about the pod)
>     Items:
>       metadata.name -> path/to/file
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8073/files/47503510a153e065d09acd5d2ece28cad1d860d9#r56556723>
>